tools/rpmuncompress.c

```diff
@@ -78,16 +84,81 @@ static char *doUncompress(const char *fn)
     return cmd;
 }
 
+/**
+ * Detect if an archive has a single top level entry, and it's a directory.
+ *
+ * @param path	path of the archive
+ * @return	1 if archive as only a directory as top level entry,
+ * 		0 if it contains multiple top level entries or a single file
+ * 		-1 on archive error
+ */
+static int singleRoot(const char *path)
+{
+	struct archive *a;
+	struct archive_entry *entry;
+	int r, ret, rootLen;
+	char *rootName;
```

Comment:
Just initialize this to NULL and then you no longer need two different exit labels.

Reply:
this because `free(NULL)` doesn't do anything?

Reply:
Yup. This is a common idiom in the codebase.